bin/inx-chronicle/src/api/auth.rs

```diff
@@ -0,0 +1,80 @@
+// Copyright 2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use async_trait::async_trait;
+use axum::{
+    extract::{FromRequest, OriginalUri},
+    headers::{authorization::Bearer, Authorization},
+    Extension, TypedHeader,
+};
+use jsonwebtoken::{DecodingKey, Validation};
+use serde::{Deserialize, Serialize};
+
+use super::{config::ApiData, ApiError};
+
+pub struct Auth;
+
+#[async_trait]
+impl<B: Send> FromRequest<B> for Auth {
+    type Rejection = ApiError;
+
+    async fn from_request(req: &mut axum::extract::RequestParts<B>) -> Result<Self, Self::Rejection> {
+        // Unwrap: Infallable
```

Comment:
Could you please use `// Panic: ...` and explain why this is infallible?

Reply:
"Infallable" IS the explanation, because `<OriginalUri as FromRequest>::Error = Infallable`

But I will add that to the comment.